src/models/Room.ts

```diff
@@ -0,0 +1,135 @@
+import { PowerLevelsEventContent } from "./events/PowerLevelsEvent";
+
+export enum RoomPreset {
+	/**
+	 * Sets:
+	 * - join_rules to `invite`
+	 * - history_visibility to `shared` 
+	 * - guest_access to `can_join`
+	 */
+	PrivateChat = "private_chat",
+	/**
+	 * Sets:
+	 * - join_rules to `invite`
+	 * - history_visibility to `shared` 
+	 * - guest_access to `can_join`
+	 * - All invitees are given the same power level as the room creator.
+	 */
+	TrustedPrivateChat = "trusted_private_chat",
+	/**
+	 * Sets:
+	 * - join_rules to `public`
+	 * - history_visibility to `shared` 
+	 * - guest_access to `forbidden`
+	 */
+	PublicChat = "public_chat",
+}
+
+export enum RoomVisibility {
+	/**
+	 * This visibility indicates that the room will be shown in the published room list. 
+	 */
+	Public = "public",
+	/**
+	 * This visibility indicates that the room will not be included in published room list.
+	 */
+	Private = "private",
+}
+
+export interface RoomCreateOptions {
+	/**
+	 * Extra keys, such as m.federate, to be added to the content of the m.room.create event.
+	 * The server will overwrite the following keys: `creator`, `room_version`.
+	 * Future versions of the specification may allow the server to overwrite other keys.
+	 */
+	creation_content?: {
+		"m.federate"?: boolean,
+		[key: string]: unknown
+	},
+	/**
+	 * A list of state events to set in the new room.
+	 * This allows the user to override the default state events set in the new room.
+	 * The expected format of the state events are an object with `type`, `state_key` and `content` keys set.
+	 * Takes precedence over events set by `preset`, but gets overridden by `name` and `topic` keys.
+	 */
+	initial_state?: {
+		/**
+		 * The content of the event.
+		 */
+		content: any,
+		/**
+		 * The state_key of the state event. Defaults to an empty string.
+		 */
+		state_key?: string,
+		/**
```

Comment:
please put a single line gap between jsdoc and the thing above it, unless it's the first thing in the object. Eg:

```typescript
{
    /**
     * The content of the event.
     */
    content: any,
    
    /**
     * The state_key of the state event. Defaults to an empty string.
     */
    state_key ? : string,
};
```

Reply:
applies throughout